src/borg/helpers/fs.py

```diff
@@ -19,6 +21,33 @@
 py_37_plus = sys.version_info >= (3, 7)
 
 
+def ensure_dir(path, name=None, mode=None, reraise=False):
+    """
+    Ensures that the dir exists with the right permissions.
+    1) Make sure the directory exists in a race-free condition
+    2) If mode is not None and the directory has been created, give the right
+    permissions
+    3) If reraise is True, catch fatal exceptions, reraise them with a pretty
+    message and the full exception output.
+    Returns True if the file has been created and has the right permissions,
+    False otherwise
+    """
+    mode = 0o777 if mode is None else mode
+    try:
+        os.makedirs(path, exist_ok=False)
+    except FileExistsError:
+        return False
```

Comment:
in what cases is this used?

Reply:
Hum, I guess if you set BORG_CACHE_DIR to an existing directory.

Reply:
Guess rather not. Also, that is the usual case except for the very first call (also: shouldn't exist_ok be True?).

But it could be that some path element happens to be a **file** and thus is in the way.

Reply:
https://github.com/borgbackup/borg/issues/5439#issuecomment-717481619

Reply:
Indeed exist_true shoud be True